classifier_load_testing/create_groups.rb

```diff
@@ -4,56 +4,66 @@
 
 num_nodes = get_arg("num_nodes", 0)
 start_at = get_arg("start_at", 1).to_i
+per_node_delay = get_arg("per_node_delay", 2).to_i
+
+create_csv = get_arg("create_csv?", 3)
+unless ["true", "false"].include?(create_csv)
+  raise "create_csv? must be 'true' or 'false'"
+end
+create_csv = (create_csv == "true") ? true : false
 
 env_group_parent_id = get_group_id('all-snow-environments')
 cv_group_parent_id = get_group_id('all-snow-classes-vars')
 
-File.write('timing.csv',"node, group, parent, elapsedTime \n")
+if create_csv
+  File.write('timing.csv',"node, group, parent, elapsedTime \n")
+end
 
-num_nodes.to_i.times do |node|
-  node = node + start_at
-  node_name = "node#{node}"
-  if (node % 100) == 0
-    puts(node_name)
+overallElapsedTime = time_block do
+  # Common helper that removes the "with_retry", "time_block" clutter.
+  # Helper returns how long it took to create the group.
+  create_group = lambda do |group|
+    retryDelay = per_node_delay.zero? ? 5 : 5 * per_node_delay
+    with_retry("Create the #{group['name']} group", 3, retryDelay) do
+      time_block do
+        sleep 1
```

Comment:
If the sleep is inside the `time_block` does that get added to the elapsed time of the api call?

Reply:
Oh shit I forgot to remove that. That was me locally testing the elapsed time. Thanks for catching

Reply:
Updated.